### ISSUE TITLE
FPS issues caused by Spatial Dash Gem

### ISSUE THREAD

**Issue by FrederikTessmann** [chylex/Hardcore-Ender-Expansion#194]
Spamming the Spatial Dash Gem causes a significant drop in frame rate that persists until relog.
Seems to be related to chylex.hee.proxy.FXClientProxy.spatialDash(FXClientProxy.java:170)

**Comment by chylex** [comment]
Even with the cooldown?

**Comment by FrederikTessmann** [comment]
Yes. The EntityEnergyClusterFX particles are not being deleted properly. I think it's caused by them never reaching their target block in EntitySoulCharmFX.onUpdate() at line 75.

**Comment by chylex** [comment]
That's an artifact of messy code structure, the particles are deleted after a set amount of time - the target block has nothing to do with it.

**Comment by FrederikTessmann** [comment]
Okay looks like I was looking at the old FX code in branch HEE1-1.7.10

**Comment by unlightednetting** [comment]
I can confirm I'm having this issue in 1.8.6

Browsing the F3 graph tool it looks like it raises root/gameRenderer/level/litParticles up to ab obscene amount - Normally it is 0.13% in my game but firing it off just seems to increase it by 5-10% each fire, regardless of whether it hits or not. FPS, imaginably, drops the more I use this item. Solutions are reconnect to the server, switch to a different dimension, restarting the game.. But it would be nice if I didn't have to do those :P

**Comment by unlightednetting** [comment]
It's been about 2 minutes and the usage on it is only going up. It's gone from 70% to 80% as I was typing the above.

**Comment by DarianLStephens** [comment]
I suppose there's no chance of any kind of fix or workaround for this issue (Besides restarting), is there?

**Comment by chylex** [comment]
No, HEE 1 is discontinued and the issue is open for HEE 2.

**Comment by DarianLStephens** [comment]
It's still a problem in HEE 1, though. Is there no way to disable the particles or force-delete them or something? It completely kills the Spatial Dash Gem as something usable.

**Comment by chylex** [comment]
Yea, there's a lot of problems in HEE 1. Switching particle setting to minimum/off may decrease the amount of particles, I can't remember.
